File: java/lsp-02-initial/src/main/java/no/bouvet/solid/lsp/services/PaymentFactory.java
package no.bouvet.solid.lsp.services;

import no.bouvet.solid.lsp.domain.PayPalPayment;
import no.bouvet.solid.lsp.domain.Payment;
import no.bouvet.solid.lsp.domain.WorldPayPayment;

public class PaymentFactory {

    public static Payment getPayment(PaymentType paymentServiceType) {

        switch (paymentServiceType) {
        case PayPal:
            return new PayPalPayment();
        case WorldPay:
            return new WorldPayPayment();
        default:
            throw new UnsupportedOperationException("No such service");
        }

    }

}

File: dotnet/Solid/Solid/LSP/Example2/Services/PaymentFactory.cs
using System;
using Solid.LSP.Example2.Domain;

namespace Solid.LSP.Example2.Services
{
    public class PaymentFactory
    {
        public static Payment GetPayment(PaymentType paymentServiceType)
        {
            switch (paymentServiceType)
            {
                case PaymentType.PayPal:
                    return new PayPalPayment();
                case PaymentType.WorldPay:
                    return new WorldPayPayment();
                default:
                    throw new NotSupportedException("No such service");
            }
        }
    }
}
